security/src/main/java/no/difi/meldingsutveksling/IntegrasjonspunktNokkel.java

```diff
@@ -0,0 +1,123 @@
+import java.security.cert.X509Certificate;
+import java.util.Enumeration;
+
+/**
+ * Class responsible for accessing the keystore for the Integrasjonspunkt.
+ *
+ * @author Glebnn Bech
+ */
+public class IntegrasjonspunktNokkel {
+
+    private final IntegrasjonspunktProperties.Keystore keystore;
+
+    public IntegrasjonspunktNokkel(IntegrasjonspunktProperties.Keystore keystore) {
+        this.keystore = keystore;
+    }
+
+    /**
+     * Loads the private key from the keystore
+     *
+     * @return the private key
+     */
+    public PrivateKey loadPrivateKey() {
+
+        PrivateKey key = null;
+        try (InputStream i = openKeyInputStream()) {
+            KeyStore keystore = KeyStore.getInstance("JKS");
+            keystore.load(i, this.keystore.getPassword().toCharArray());
+
+            Enumeration aliases = keystore.aliases();
+            while (aliases.hasMoreElements()) {
+                String alias = (String) aliases.nextElement();
+                boolean isKey = keystore.isKeyEntry(alias);
+                if (isKey && alias.equals(this.keystore.getAlias())) {
+                    key = (PrivateKey) keystore.getKey(alias, this.keystore.getPassword().toCharArray());
+                }
+            }
+            if (key == null) {
+                throw new RuntimeException("no key with alias " + this.keystore.getAlias() + " found in the keystore " +
+                        this.keystore.getPath());
+            }
+            return key;
+        } catch (CertificateException | KeyStoreException | IOException | NoSuchAlgorithmException | UnrecoverableKeyException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public KeyPair getKeyPair() {
+
+        KeyPair result = null;
+        try (InputStream i = openKeyInputStream()) {
+            KeyStore keystore = KeyStore.getInstance("JKS");
+            keystore.load(i, this.keystore.getPassword().toCharArray());
+            Enumeration aliases = keystore.aliases();
+            for (; aliases.hasMoreElements();) {
+                String alias = (String) aliases.nextElement();
+                boolean isKey = keystore.isKeyEntry(alias);
+                if (isKey && alias.equals(this.keystore.getAlias())) {
+                    PrivateKey key = (PrivateKey) keystore.getKey(alias, this.keystore.getPassword().toCharArray());
+                    X509Certificate c = (X509Certificate) keystore.getCertificate(alias);
+                    result = new KeyPair(c.getPublicKey(), key);
+                    break;
+                }
+            }
+            if (result == null) {
+                throw new RuntimeException("no key with alias " + this.keystore.getAlias() + " found in the keystore " +
+                        this.keystore.getPath());
+            }
+        } catch (CertificateException | KeyStoreException | IOException | NoSuchAlgorithmException | UnrecoverableKeyException e) {
+            throw new RuntimeException(e);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00112)